build_utils/bump_changelog.py

```diff
@@ -0,0 +1,122 @@
+#! /usr/bin/env python
+
+"""Bump the current version number in CHANGELOG.rst
+
+Unreleased changes will be moved into a section for the new version number
+"""
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from pathlib import Path
+import re
+
+from toolz.itertoolz import partition
+
+REPOSITORY_ADDRESS = "https://github.com/pace-neutrons/Euphonic"
+LINK_STRING = "`{tag} <" + REPOSITORY_ADDRESS + "/compare/{previous_tag}...{compare_tag}>`_"
```

Comment:
Again, not sure if cleaner or more obscure..?
```suggestion
LINK_STRING = f"`{{tag}} <{REPOSITORY_ADDRESS}/compare/{{previous_tag}}...{{compare_tag}}>`_"
```